superset/assets/javascripts/explore/exploreUtils.js

```diff
@@ -1,7 +1,34 @@
 /* eslint camelcase: 0 */
 import URI from 'urijs';
 
-export function getExploreUrl(form_data, endpointType = 'base', force = false,
+export function getURIDirectory(form_data, endpointType = 'base') {
+  // Building the directory part of the URI
+  let directory = '/superset/explore/';
+  if (['json', 'csv', 'query'].indexOf(endpointType) >= 0) {
+    directory = '/superset/explore_json/';
+  }
+  const [datasource_id, datasource_type] = form_data.datasource.split('__');
+  directory += `${datasource_type}/${datasource_id}/`;
+
+  return directory;
+}
+
+export function getExploreLongUrl(form_data, endpointType) {
+  if (!form_data.datasource) {
+    return null;
+  }
+
+  const uri = new URI('/');
+  const directory = getURIDirectory(form_data, endpointType);
+  const search = uri.search(true);
+  search.form_data = JSON.stringify(form_data);
+  if (endpointType === 'standalone') {
+    search.standalone = 'true';
+  }
+  return uri.directory(directory).search(search).toString();
+}
+
+export function createExplore(form_data, endpointType = 'base', force = false,
```

Comment:
same thoughts about an object signature (less error-prone and calls are more readable), also could we use camelCase? `({ formData, endpointType = 'base', force = false, currUrl = null, requestParams = {} })`

Reply:
could we make this method name more clear? like `getExploreUrlAndPayload` it took me a while to understand what `getExplore` meant.

Reply:
fixed. used object signatures for `getExploreUrlAndPayload` method